File: src/main/java/com/example/today_i_baekjoon/repository/HolidayRepository.java
package com.example.today_i_baekjoon.repository;

import com.example.today_i_baekjoon.domain.Holiday;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDate;
import java.util.Optional;

public interface HolidayRepository extends JpaRepository<Holiday, Long> {

    Optional<Holiday> findByDate(LocalDate date);
}

File: back/src/main/kotlin/com/gidskql6671/today_i_baekjoon/repository/HolidayRepository.kt
package com.gidskql6671.today_i_baekjoon.repository

import com.gidskql6671.today_i_baekjoon.domain.Holiday
import org.springframework.data.jpa.repository.JpaRepository
import java.time.LocalDate

interface HolidayRepository: JpaRepository<Holiday, Long> {

    fun findByDate(date: LocalDate): Holiday?

    fun findAllByDateBetween(start: LocalDate, end: LocalDate): List<Holiday>

    fun existsByDate(date: LocalDate): Boolean

    fun deleteByDateIn(date: List<LocalDate>)
}